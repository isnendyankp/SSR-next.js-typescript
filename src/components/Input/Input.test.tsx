/*
    this is a test file for the input component
*/
import { render, screen } from '@testing-library/react';
import Input from '.';

describe('Input component unit testing', () => {
    // this test for input component with explanation should render input with placeholder
    it('should render input with placeholder', () => {
        render(<Input placeholder="Search" />);
        const inputElement = screen.getByPlaceholderText(/search/i);
        expect(inputElement).toMatchSnapshot();
    });
});
