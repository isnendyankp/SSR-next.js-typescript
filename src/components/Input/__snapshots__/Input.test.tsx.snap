// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component unit testing should render input with placeholder 1`] = `
<input
  class="undefined block flex-1 border-2 bg-transparent py-1.5 pl-1 text-gray-900 placeholder:text-gray-400 focus:ring-0 sm:text-sm sm:leading-6"
  placeholder="Search"
  type="text"
/>
`;
